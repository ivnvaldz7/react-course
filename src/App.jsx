import PropTypes from "prop-types";

export const App = ({ title, subtitle, name }) => {
  return (
    <>
      <h1>{title}</h1>
      <h2>{subtitle}</h2>
      <h4>{`Con ${name}`}</h4>
    </>
  );
};
App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string.isRequired,
};
App.defaultProps = {
  subtitle: "No existe texto",
  name: "No existe texto",
};
