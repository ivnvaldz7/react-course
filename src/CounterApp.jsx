import PropTypes from "prop-types";
import { useState } from "react";
export const CounterApp = ({ value, title }) => {
  const [counter, setCounter] = useState(value);
  return (
    <>
      <h1>{title}</h1>
      <p>{counter}</p>
      <button onClick={() => setCounter(counter + 1)}> +1</button>
      <button onClick={() => setCounter(counter - 1)}> -1</button>
      <button onClick={() => setCounter(value)}> reset</button>
    </>
  );
};
CounterApp.propTypes = {
  value: PropTypes.number,
  title: PropTypes.string.isRequired
};
CounterApp.defaultProps = {
  title: "Title of Counter"
}
