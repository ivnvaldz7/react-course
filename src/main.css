body,html{
    background-color: #21232A;
    color: aliceblue;
    font-family: helvetica, arial;
    font-size: 1rem;
    padding: 70px;
}
h4{
    font-family: 'Italic';
}
h1{
    font-weight: Bold;
}