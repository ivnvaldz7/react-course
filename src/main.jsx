import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";
import {App} from './App';
import "./main.css";
const root = ReactDOM.createRoot(document.getElementById("root"));

root.render(
  <React.StrictMode>
    <CounterApp value={10} />
    {/* <App title="Curso de React" subtitle='De 0 a experto' name='Fernando Herrera'/> */}
  </React.StrictMode>
);
