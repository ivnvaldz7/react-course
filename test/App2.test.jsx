/** * @jest-environment jsdom */

import { render, screen } from "@testing-library/react"
import { App } from "../src/App"


describe('Prueba en </App> ', () => { 
  const title = 'Curso de React'
  const subtitle = 'Con Fernando Herrera'
   test('Debe hacer match con snapshot', () => { 
    const {conatiner} = render(<App title={title}/>)
    expect(conatiner).toMatchSnapshot()
    })
    test('debe retorar un title', () => { 
      render(<App title={title}/>)
      expect(screen.getByText(title)).toBeTruthy()
     })    
 })