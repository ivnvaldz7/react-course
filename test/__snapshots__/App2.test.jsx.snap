// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en </App>  Debe hacer match con snapshot 1`] = `undefined`;
